### PR TITLE
add CMakefile for nxp and fix build problems

### PR DESCRIPTION
**Pull request by qiutongs**
Description
-----------

- add CMakefile for nxp mcuxpresso
- exclude cborpretty.c because it includes a bad header "inttype.h"
- fix multi-definition problem in demo

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

### CODE REVIEW

File: vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+        "${AFR_KERNEL_DIR}/portable/MemMang/heap_4.c"
+)
+
+set(kernel_inc_dirs
+    ${lpc54018_includes}
+    "${AFR_KERNEL_DIR}/portable/GCC/ARM_CM4F"
+    "${AFR_KERNEL_DIR}/portable/CCS/ARM_CM3"
+    "${lpc54018_app_dir}/config_files"
+    "${lpc54018_app_dir}/application_code/nxp_code"
+    "$<IF:${AFR_IS_TESTING},${AFR_TESTS_DIR},${AFR_DEMOS_DIR}>/include"
```
Comment (mradulan) [inline vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: don't we need `${lpc54018_ports_dir}/<module>/include`? Like where is aws_wifi.h included?

Comment (qiutongs) [reply vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: It is the common path: /amazon-freertos/demos/include

Not board specific as lpc54018_ports_dir

Comment (mradulan) [reply vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: it is in `AFR_Root/modules/ports/wifi/include` but I think it is included in the modules CMakeListst.txt

File: vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,317 @@
+cmake_minimum_required(VERSION 3.13)
```
Comment (tgsong) [inline vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: We don't need this.

File: vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+        $<$<NOT:$<COMPILE_LANGUAGE:ASM>>:${c_flags}>
+        -specs=redlib.specs -MMD -MP
+)
+
+# Assembler flags
+target_compile_options(
+    AFR::compiler::mcu_port        
+    INTERFACE
+        ${common_flags}
+        $<$<NOT:$<COMPILE_LANGUAGE:C>>:${asm_flags}>
```
Comment (tgsong) [inline vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: Change this to `$<$<COMPILE_LANGUAGE:ASM>:${asm_flags}>`

File: vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+)
+
+# PKCS11
+afr_mcu_port(pkcs11)
+target_sources(
+    AFR::pkcs11::mcu_port
+    INTERFACE 
+        "${lpc54018_ports_dir}/pkcs11/aws_pkcs11_pal.c"
+        "${lpc54018_ports_dir}/pkcs11/hw_poll.c"
+        "${AFR_MODULES_PORTS_DIR}/pkcs11/mbedtls/aws_pkcs11_mbedtls.c"
```
Comment (tgsong) [inline vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: We already defined a target that uses this default implementation, so you don't need to specify it here. Instead, use
```cmake
target_link_libraries(
    AFR::pkcs11::mcu_port
    INTERFACE pkcs11_mbedtls
)

File: vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+endif()
+
+afr_glob_src(board_code_src DIRECTORY "${lpc54018_app_dir}/application_code/nxp_code")
+afr_glob_src(config_files_src DIRECTORY "${lpc54018_app_dir}/config_files")
+
+add_executable(${exe_target}
+    ${board_code_src}
+    ${config_files_src}
+    "${lpc54018_app_dir}/application_code/main.c"
+    # It is unknown why hw_poll.c has to be included here
```
Comment (tgsong) [inline vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: So if it's compiled into pkcs11 static library, the linker cannot find the function symbols defined in it?

File: vendors/nxp/LPC54018/mcuxpresso/aws_demos_Debug.ld [hunk 6]
```diff
@@ -0,0 +1,243 @@
+/*
```
[the remark this answers is not in the document]
Comment (qiutongs) [reply vendors/nxp/LPC54018/mcuxpresso/aws_demos_Debug.ld]: No. We need to reach out to NXP.

File: vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt (hunk 2 above)
Comment (qiutongs) [reply vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: ok, removed.

File: vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt (hunk 3 above)
Comment (qiutongs) [reply vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: ok

File: vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt (hunk 4 above)
Comment (qiutongs) [reply vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: You mean "AFR::pkcs11_mbedtls" right? OK

File: vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt (hunk 5 above)
Comment (qiutongs) [reply vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: Yeah...

amazon-freertos/modules/libraries/3rdparty/mbedtls/library/entropy.c:120: undefined reference to `mbedtls_hardware_poll'

File: vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt (hunk 1 above)
Comment (mradulan) [inline vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: change AFR_TESTS_DIR --> AFR_TEST_DIR

Comment (qiutongs) [reply vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt]: Good catch. Thanks

**Pull request merged**